### PR TITLE
gcc 13.1.0

### PR DESCRIPTION
**Pull request by fxcoudert**
Testing the GCC 13 release candidate

### CODE REVIEW

**Comment by fxcoudert** [comment]
```
  gcc:
    * Formula refers to a GitHub issue or pull request that is closed: https://github.com/Linuxbrew/brew/issues/724
  Error: 1 problem in 1 formula detected
  Error: no implicit conversion of Symbol into Integer
```

Not sure why linking to a closed issue is forbidden. And no idea where the second error comes from.

**Comment by carlocab** [comment]
> Not sure why linking to a closed issue is forbidden.

Well, it was useful to clean up a handful of stale issue references here and there, but we also now have a bunch of owner/repo#xyz references that used to be URLs (to evade the audit), so it's a mixed bag, really.

Thoughts on this audit, @Homebrew/core?

> And no idea where the second error comes from.

Seems like an audit bug.

**Comment by p-linnane** [comment]
> Thoughts on this audit, @Homebrew/core?

Conceptually, I get it. But I think it's been causing more harm than good. Lots of erasing knowledge of why we are doing certain things they way we are just because a related PR or issue is closed.

**Comment by MikeMcQuaid** [comment]
> > Not sure why linking to a closed issue is forbidden.
> 
> Well, it was useful to clean up a handful of stale issue references here and there, but we also now have a bunch of owner/repo#xyz references that used to be URLs (to evade the audit), so it's a mixed bag, really.

I think we should revert this audit and restore the things that have been removed due to it. A non-zero number of maintainers actively dislike it and these sort of workarounds to evade the audit make it actively harmful.

**Comment by carlocab** [comment]
> I think we should revert this audit and restore the things that have been removed due to it. A non-zero number of maintainers actively dislike it and these sort of workarounds to evade the audit make it actively harmful.

Homebrew/brew#15284

**Comment by fxcoudert** [comment]
Looking very good!

- missing one `r` revision bump
- rebottling `mpv` for unrelated issue on Linux: https://github.com/Homebrew/homebrew-core/actions/runs/4771739269 (missing linking to libplacebo.so.264)

**Comment by carlocab** [comment]
> * missing one `r` revision bump

Kinda weird that `r` revision bump seems to only be needed on ARM?

**Comment by fxcoudert** [comment]
> Kinda weird that r revision bump seems to only be needed on ARM?

Yes, but it really is needed: apparently that R has recorded an absolute library path.

**Comment by carlocab** [comment]
> Yes, but it really is needed: apparently that R has recorded an absolute library path.

It's in a text file, though: `lib/R/etc/Makeconf`. The difference is that ARM `-lemutls_w` but Intel doesn't, and `libemutls_w.a` is in `lib/gcc/current/gcc/*-apple-darwin*/#{version.major}`.

I wonder if we can/should give that path the `current` treatment so it doesn't break on GCC version bumps. The path only seems to contain static libraries and headers.

File: Formula/gcc@12.rb
Comment (carlocab) [inline Formula/gcc@12.rb:1]: Needs to be added to the versioned keg-only allowlist

**Review by carlocab — APPROVED** [review]

**Comment by fxcoudert** [comment]
Linux failures:
- CP2K (not sure what the issue is, but the code was running fine, so not worrying)
- root linkage, bottle rebuilt started: https://github.com/Homebrew/homebrew-core/actions/runs/4828325206

**Review by SMillerDev — APPROVED** [review]

**Comment by fxcoudert** [comment]
`octave` does not want to bottle on Ventura, because it depends on `qt@5`, which has no Ventura bottles.

**Review by carlocab — APPROVED** [review]
Ok with merging this. Ventura dependent issues seem unlikely given that CI is all green elsewhere.

**Comment by fxcoudert** [comment]
As I understand, merging in current state is against policy: it would remove the Ventura octave bottles.

**Review by carlocab — APPROVED** [review]

**Comment by p-linnane** [comment]
Linux failures:
```
Error: 2 failed steps!
brew test --retry --verbose cp2k
brew linkage --test root
```
We can follow up on these afteward. Thank you @fxcoudert for all the hard work here!

**Pull request merged**